src/coreclr/jit/redundantbranchopts.cpp

```diff
@@ -534,30 +517,14 @@ bool Compiler::optJumpThread(BasicBlock* const block, BasicBlock* const domBlock
     //
     for (BasicBlock* const predBlock : block->PredBlocks())
     {
-        if (predBlock->bbJumpKind == BBJ_SWITCH)
-        {
-            // Skip over switch preds, they will continue to flow to block.
-            //
-            continue;
-        }
-
-        const bool isTruePred =
-            ((predBlock == domBlock) && (trueSuccessor == block)) || optReachable(trueSuccessor, predBlock, domBlock);
-        const bool isFalsePred =
-            ((predBlock == domBlock) && (falseSuccessor == block)) || optReachable(falseSuccessor, predBlock, domBlock);
```

Comment:
Is `optReachable` still used anywhere after this change?

Reply:
The initial classification loop still calls it.

Reply:
Just to check, is that needed over `fgReachable` to be able to check for reachability that does not pass through `domBlock`? The other reason was that the flow graph was being modified in the second loop, but I guess that reason is gone now.

Reply:
I think it still offers benefit because prior RBO's and jump threads may impact reachability. But not 100% sure.

Reply:
Ah, the classification is still running multiple times, I see.